src/DemoFile/BitBuffer.cs

```diff
@@ -28,7 +41,7 @@ public uint ReadUBits(int numBits)
     {
```

Comment:
Can you please add a safeguard here, so it's easier to catch similar bugs :
```
#if DEBUG
if (numBits > 32 || numBits < 0)
    throw ...;
#endif
```

Reply:
This will throw as the code is already written

Reply:
Maybe I am missing something, but if the code goes into `else` branch and if  `numBits` is 50 and `_bitsAvail` is 32, it will not throw, but instead decrease `numBits` to 18.

Anyways, it's not intuitive as it is, so adding `Debug.Assert()` would not hurt.